js/spec/cursor_spec.js

```diff
@@ -0,0 +1,364 @@
+/**
+ * Copyright 2017 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ------------------------------------------------------------------------------
+ */
+
+'use strict'
+
+const assert = require('assert')
+const querystring = require('querystring')
+const request = require('superagent')
+const mock = require('superagent-mocker')(request)
+
+const {Cursor} = require('../lib')
+
+describe('Cursor', () => {
+  beforeEach(() => {
+    mock.clearRoutes()
+  })
+
+  describe('.all()', () => {
+    it('should return all values of a query, with no pages', () => {
+      mock.get('/paging_endpoint', (req) => ({
+        body: {
+          data: [{x: 1, y: 1}, {x: 5, y: 5}],
+          paging: {
+            start_index: 0,
+            total_count: 2
+          }
+        }
+      }))
+
+      let cursor = new Cursor('/paging_endpoint')
+
+      return cursor.all().then(objs => {
+        assert.deepEqual([
+          {x: 1, y: 1},
+          {x: 5, y: 5}
+        ],
```

Comment:
It follows the pattern of all the tests, where the data is "parsed", though in this case, it is parsed with the identity function.